astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -76,10 +95,30 @@ def defer(self, last_log_time: Optional[DateTime] = None, **kwargs: Any) -> None
             method_name=self.trigger_reentry.__name__,
         )
 
-    def execute(self, context: Context) -> None:  # noqa: D102
+    def execute(self, context: Context) -> Any:  # noqa: D102
         self.pod_request_obj = self.build_pod_request_obj(context)
         self.pod: k8s.V1Pod = self.get_or_create_pod(self.pod_request_obj, context)
-        self.defer()
+        container_state = self.define_container_state()
+        pod_status = self.pod.status.phase
+        if KubernetesPodTrigger.should_wait(pod_status, container_state):
+            self.defer()
+            return
+
+        if container_state == ContainerState.TERMINATED:
```

Comment:
does pod TERMINATED state mean always a success?

Reply:
Yes, I think so. This is how we decide the state in OSS airflow https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/triggers/pod.py#L135 as well